stdlib/public/core/Codable.swift

```diff
@@ -3884,153 +3809,134 @@ public extension UnkeyedEncodingContainer {
 
 // Default implementations for decode(_:) in terms of decodeIfPresent(_:)
 public extension UnkeyedDecodingContainer {
+    @_semantics("optimize.sil.specialize.generic.never")
+    @_semantics("optimize.sil.specialize.generic.partial.never")
+    @inline(never)
+    internal func decodeError<T: Decodable>(_ type: T.Type) throws -> T {
```

Comment:
Same comment here. Why pretend to return a `T` when we can just return the error?